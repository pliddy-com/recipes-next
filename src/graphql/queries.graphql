# Query for Homepage RecipeCollection (all recipes)

query homePageQuery {
  recipeCollection(order: title_ASC) {
    total
    items {
      ...RecipeSummary
    }
  }
}

# Query for Recipe collection based on filter

query recipeCollection($where: RecipeFilter) {
  recipeCollection(where: $where, order: title_ASC) {
    total
    items {
      ...RecipeSummary
    }
  }
}

# Query for Recipe page

query recipePage($where: RecipeFilter) {
  recipeCollection(where: $where, order: title_ASC, limit: 1) {
    total
    items {
      ...RecipeDefault
    }
  }
}

# Query for List Page RecipeCollection (all recipes matching tag)

query listPageQuery($where: TagFilter) {
  tagCollection(where: $where, limit: 1) {
    total
    items {
      slug
      title
      linkedFrom {
        recipeCollection(limit: 24) {
          total
          items {
            ...RecipeSummary
          }
        }
      }
    }
  }
}

# Query for recipe slugs

query recipeSlugsCollection($where: RecipeFilter) {
  recipeCollection(where: $where) {
    items {
      slug
    }
  }
}

# query for category slugs

query categorySlugsCollection($where: TagFilter) {
  tagCollection(where: $where) {
    items {
      slug
    }
  }
}

# query for category slugs

query tagSlugsCollection($where: TagFilter) {
  tagCollection(where: $where) {
    items {
      slug
      linkedFrom {
        recipeCollection {
          total
        }
      }
    }
  }
}

# query for Taxonomy with Category & Subcategory levels

query taxonomyCollection($where: TaxonomyFilter) {
  taxonomyCollection(where: $where, limit: 1) {
    items {
      ...TaxonomyDefault

      tag {
        ...TagDefault
      }

      childrenCollection(limit: 24) {
        total
        items {
          ... on Tag {
            ...TagDefault

            linkedFrom {
              recipeCollection(limit: 1) {
                total
              }
            }
          }
          ...TaxonomyWithChildren
        }
      }
    }
  }
}
