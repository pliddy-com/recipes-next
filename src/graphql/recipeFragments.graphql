# Fragments with default fields for Recipe-specific sections:
#   * Ingredients
#   * Instructions
#   * Image
#   * Recipe

fragment IngredientsDefault on IngredientSection {
  sys {
    id
  }
  title
  slug
  label
  ingredientList
}

fragment InstructionsDefault on InstructionSection {
  sys {
    id
  }
  title
  slug
  label
  instructionList
}

fragment ImageDefault on Asset {
  sys {
    id
  }
  __typename
  title
  description
  contentType
  fileName
  size
  url
  height
  width
}

fragment DescriptionDefault on Recipe {
  description {
    json
    links {
      entries {
        inline {
          __typename
          sys {
            id
            __typename
          }
        }
        hyperlink {
          __typename
          sys {
            id
            __typename
          }
        }
        block {
          __typename

          sys {
            id
            __typename
          }
        }
      }
      assets {
        hyperlink {
          __typename
          title
        }
        block {
          __typename
          title
        }
      }
    }
  }
}

fragment RecipeDefault on Recipe {
  sys {
    id
  }
  __typename
  title
  slug
  description {
    json
  }
  abstract
  image {
    ...ImageDefault
  }
  ingredientsCollection(limit: 10) {
    items {
      ...IngredientsDefault
    }
  }
  equipment
  instructionsCollection(limit: 10) {
    items {
      ...InstructionsDefault
    }
  }
  notes
  tagsCollection(limit: 10) {
    items {
      ...TagDefault
    }
  }
}
