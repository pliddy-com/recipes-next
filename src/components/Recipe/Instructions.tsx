import Avatar from '@mui/material/Avatar';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';
import ListItemAvatar from '@mui/material/ListItemAvatar';
import ListItemText from '@mui/material/ListItemText';
import Stack from '@mui/material/Stack';
import Typography from '@mui/material/Typography';

import { InstructionsDefaultFragment } from 'types/generated/graphql';

interface InstructionsProps {
  sections?: InstructionsDefaultFragment[];
}

const Instructions = ({ sections }: InstructionsProps) => {
  if (!sections) return null;

  // use num to set starting value for each instructionList section
  // so numbering is continuous across list sections
  let num = 1;

  return (
    <Stack direction="column" spacing={1}>
      <Typography variant="h2" gutterBottom>
        Instructions
      </Typography>
      {sections &&
        sections.map((section) => {
          const { label, instructionList } = section ?? {};
          const instructions = instructionList;
          return (
            <Stack key={label} direction="column" spacing={0}>
              <Typography variant="h3">{label}</Typography>

              <List className="recipeList">
                {instructions &&
                  instructions.map((instruction) => (
                    <ListItem key={`instruction-${num}`} disableGutters>
                      <ListItemAvatar>
                        <Avatar color="secondary">{num++}</Avatar>
                      </ListItemAvatar>
                      <ListItemText primary={instruction} />
                    </ListItem>
                  ))}
              </List>
            </Stack>
          );
        })}
    </Stack>
  );
};

export default Instructions;
