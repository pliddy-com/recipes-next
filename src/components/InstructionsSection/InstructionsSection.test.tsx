// import testing-library methods
import { render, screen } from '@testing-library/react';

// add custom jest matchers from jest-dom
import '@testing-library/jest-dom';

// import the component to test
import Instructions from 'components/InstructionsSection/InstructionsSection';

import { InstructionsDefaultFragment } from 'types/generated/graphql';

describe('in Instructions', () => {
  const expectedTitle = 'Instructions';

  it('renders the instructions section if there is content', () => {
    const sections: InstructionsDefaultFragment[] = [
      {
        sys: {
          id: '2xmjf2j2BopShF9fUli4BV',
          __typename: 'Sys',
        },
        title: 'Sole Meunière, Setup',
        slug: 'sole-meuniere-setup',
        label: 'Setup',
        instructionList: ['Heat saute pan over medium-high heat.'],
        __typename: 'InstructionSection',
      },
    ];

    const expectedSubtitle = sections?.[0].label;
    const expectedItem = sections?.[0].instructionList?.[0];

    render(<Instructions sections={sections} />);

    const title = screen.queryByText(expectedTitle);
    const subtitle = expectedSubtitle && screen.queryByText(expectedSubtitle);
    const item = expectedItem && screen.queryByText(expectedItem);

    expect(title).toBeInTheDocument();
    expect(subtitle).toBeInTheDocument();
    expect(item).toBeInTheDocument();
  });

  it('does not render the ingredients section if there is no content', () => {
    const sections = undefined;

    render(<Instructions sections={sections} />);

    const title = screen.queryByText(expectedTitle);

    expect(title).toBeNull();
  });

  it('does not render the instructions section if there are missing properties content', () => {
    const sections: InstructionsDefaultFragment[] = [
      {
        sys: {
          id: '2xmjf2j2BopShF9fUli4BV',
          __typename: 'Sys',
        },
        title: 'Sole Meunière, Setup',
        slug: 'sole-meuniere-setup',
        label: null,
        instructionList: null,
        __typename: 'InstructionSection',
      },
    ];

    const expectedSubtitle = sections?.[0].label;

    render(<Instructions sections={sections} />);

    const subtitle = expectedSubtitle && screen.queryByText(expectedSubtitle);
    expect(subtitle).toBeNull();
  });

  it('does not render the instructions section if there are missing properties content', () => {
    const sections = [null];

    const { container } = render(<Instructions sections={sections} />);

    const list = container.querySelector('.recipeList');
    expect(list).toBeNull();
  });
});

/*
{
    const sections =  [
        {
            "sys": {
                "id": "2xmjf2j2BopShF9fUli4BV",
                "__typename": "Sys"
            },
            "title": "Sole Meunière, Setup",
            "slug": "sole-meuniere-setup",
            "label": "Setup",
            "instructionList": [
                "Heat saute pan over medium-high heat."
            ],
            "__typename": "InstructionSection"
        },
        {
            "sys": {
                "id": "7MvCMbZIzxJb7j9t9TYor",
                "__typename": "Sys"
            },
            "title": "Sole Meunière, Prep",
            "slug": "sole-meuniere-prep",
            "label": "Prep",
            "instructionList": [
                "Sprinkle salt and pepper over both sides of the filets.",
                "Dredge each filet in flour and shake off any excess flour."
            ],
            "__typename": "InstructionSection"
        },
        {
            "sys": {
                "id": "6ErpZlcHdJi7xt1iZtiRrs",
                "__typename": "Sys"
            },
            "title": "Sole Meunière, Sauté",
            "slug": "sole-meuniere-saute",
            "label": "Sauté",
            "instructionList": [
                "Heat oil in the pan and add butter until it is melted and stops foaming. ",
                "Saute filets until brown (3-5 minutes), then flip and brown the other side (3-4 min)",
                "Transfer filets to the serving plate."
            ],
            "__typename": "InstructionSection"
        },
        {
            "sys": {
                "id": "3gNHPOVv2fmA12Pz0XQXER",
                "__typename": "Sys"
            },
            "title": "Sole Meunière, Sauce",
            "slug": "sole-meuniere-sauce",
            "label": "Sauce",
            "instructionList": [
                "Sprinkle filets with fresh chopped parsley.",
                "In a light-colored sauce pan, melt 2 Tbsp of butter, stirring until brown.",
                "Remove browned butter from the heat and toss in lemon and capers.",
                "Pour beurre noisette over filets to sizzle the parsley."
            ],
            "__typename": "InstructionSection"
        }
    ]
}
*/
