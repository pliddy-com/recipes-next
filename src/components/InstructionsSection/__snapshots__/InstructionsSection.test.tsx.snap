// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`in InstructionsSection when there is properly formatted content it renders the instructions section if there is content 1`] = `
<div>
  <div
    class="css-1okj3ks-MuiStack-root"
  >
    <h2
      class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom css-i1hrn3-MuiTypography-root"
    >
      Instructions
    </h2>
    <div
      class="css-nen11g-MuiStack-root"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
      >
        Setup
      </h3>
      <ul
        class="MuiList-root MuiList-padding recipeList css-h4y409-MuiList-root"
      >
        <li
          class="MuiListItem-root MuiListItem-padding css-1tnift2-MuiListItem-root"
        >
          <div
            class="MuiListItemAvatar-root css-1e9lk82-MuiListItemAvatar-root"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
              color="secondary"
            >
              1
            </div>
          </div>
          <div
            class="MuiListItemText-root css-tlelie-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
            >
              Heat saute pan over medium-high heat.
            </span>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
